app/js/instances/instanceModel.js

```diff
@@ -77,18 +83,76 @@ function instanceModel(dockerService) {
             this.emitChange();
         },
 
+        imageShowAll: function (r) {
+            this.showAllImages = r;
+        },
+
+        searchImages: function(r) {
+            this.foundImages = r.response;
+            this.emitChange();
+        },
+
+        pullImage: function(r) {
+            var imageTag = r.meta.query.fromImage;
+
+            if (r.pattern === '{error}') {
+                alertService.create({
+                    message: statusUpdate.error,
+                    type: 'danger'
+                });
+                this.loadingImages = _.omit(this.loadingImages, imageTag);
+                return;
+            }
+
+            // right now we have to guess when the image pull has finished
+            // https://github.com/jimhigson/oboe.js/issues/44
+            if (_.startsWith(r.response.status, 'Status: Downloaded newer image') ||
+                _.startsWith(r.response.status, 'Status: Image is up to date')) {
+
+                this.loadingImages = _.omit(this.loadingImages, imageTag);
+                dockerService.d('images.list', {
+                    host: r.host,
+                    query: {all: this.showAllImages}
+                });
```

Comment:
Yeah that's fair.  My assumption was that the emit change would be fired later down the road as a result of calling ```image.list``` on the docker service.  Probably better to be safe than sorry.